# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1327886339 -3600
#      Mon Jan 30 02:18:59 2012 +0100
# Node ID ba360324035e330eb3c7266327fc4d24a67b1f9a
# Parent  c22fa57226b2600637cee6a264ebd76118837474
Fixed bug with redrawing the terminal when widget is being shown.

diff --git a/libqterminal/QTerminal.cpp b/libqterminal/QTerminal.cpp
--- a/libqterminal/QTerminal.cpp
+++ b/libqterminal/QTerminal.cpp
@@ -132,16 +132,23 @@ void QTerminal::setReadOnly(bool readonl
 void QTerminal::focusInEvent(QFocusEvent *focusEvent)
 {
     Q_UNUSED(focusEvent);
     m_sessionView->updateImage();
     m_sessionView->repaint();
     m_sessionView->update();
 }
 
+void QTerminal::showEvent(QShowEvent *)
+{
+    m_sessionView->updateImage();
+    m_sessionView->repaint();
+    m_sessionView->update();
+}
+
 void QTerminal::resizeEvent(QResizeEvent*)
 {
     m_sessionView->resize(this->size());
     m_sessionView->updateImage();
     m_sessionView->repaint();
     m_sessionView->update();
 }
 
diff --git a/libqterminal/QTerminal.h b/libqterminal/QTerminal.h
--- a/libqterminal/QTerminal.h
+++ b/libqterminal/QTerminal.h
@@ -44,16 +44,17 @@ signals:
     void finished();
 
 public slots:
     void copyClipboard();
     void pasteClipboard();
         
 protected:
     void focusInEvent(QFocusEvent *focusEvent);
+    void showEvent(QShowEvent *);
     virtual void resizeEvent(QResizeEvent *);
     
 protected slots:
     void sessionFinished();        
     
 private:
     void initialize();
     void connectToPty();
